benchmark/torch/ppo/train.py

```diff
@@ -12,151 +12,143 @@
-                    MAX_GRAD_NROM)
-
-    agent = MujocoAgent(algorithm, device)
-
-    rollouts = RolloutStorage(NUM_STEPS, env.observation_space.shape[0],
-                              env.action_space.shape[0])
-
-    obs = env.reset()
-    rollouts.obs[0] = np.copy(obs)
-
-    episode_rewards = deque(maxlen=10)
-
-    num_updates = int(args.train_total_steps) // NUM_STEPS
-    for j in range(num_updates):
-        for step in range(NUM_STEPS):
-            # Sample actions
-            value, action, action_log_prob = agent.sample(rollouts.obs[step])
-
-            # Obser reward and next obs
-            obs, reward, done, info = env.step(action)
-            # get training rewards from info['episode']['r']
-            if done:
-                episode_rewards.append(info['episode']['r'])
-
-            # If done then clean the history of observations.
-            masks = torch.FloatTensor([[0.0]] if done else [[1.0]])
-            bad_masks = torch.FloatTensor([[0.0]] if 'bad_transition' in info.
-                                          keys() else [[1.0]])
-            rollouts.append(obs, action, action_log_prob, value, reward, masks,
-                            bad_masks)
-
-        next_value = agent.value(rollouts.obs[-1])
-
-        value_loss, action_loss, dist_entropy = agent.learn(
-            next_value, GAMMA, GAE_LAMBDA, PPO_EPOCH, BATCH_SIZE, rollouts)
-
-        rollouts.after_update()
-
-        if j % LOG_INTERVAL == 0 and len(episode_rewards) > 1:
-            total_num_steps = (j + 1) * NUM_STEPS
-            print(
-                "Updates {}, num timesteps {},\n Last {} training episodes: mean/median reward {:.1f}/{:.1f}, min/max reward {:.1f}/{:.1f}\n"
-                .format(j, total_num_steps, len(episode_rewards),
-                        np.mean(episode_rewards), np.median(episode_rewards),
-                        np.min(episode_rewards), np.max(episode_rewards),
-                        dist_entropy, value_loss, action_loss))
-
-        if (args.test_every_steps is not None and len(episode_rewards) > 1
-                and j % args.test_every_steps == 0):
-            # get current ob_rms from training env and use it for evaluation
-            ob_rms = get_ob_rms(env)
-            eval_mean_reward = evaluate(agent, ob_rms)
-
-            summary.add_scalar('ppo/mean_validation_rewards', eval_mean_reward,
-                               (j + 1) * NUM_STEPS)
+    logger.info("------------------- PPO ---------------------")
+    logger.info('Env: {}, seed: {}'.format(args.env, args.seed))
+    logger.info("---------------------------------------------")
+
+    config = Config['mujoco'] if args.continuous_action else Config['atari']
+    if args.env_num:
+        config['env_num'] = args.env_num
+    config['batch_size'] = int(config['env_num'] * config['step_nums'])
+    config['minibatch_size'] = int(
+        config['batch_size'] // config['num_minibatches'])
```

Comment:
I think `minibatch_size` can be computed in agent.py, and it's not necessary to mix rollout config and alg config